### PR TITLE
throw invalid parameter errors in `crypto.scryptSync`

### PR DESCRIPTION
**Pull request by dylan-conway**

### CODE REVIEW

File: src/bun.js/bindings/bindings.zig [hunk 1]
```diff
@@ -2414,6 +2414,19 @@ pub const JSGlobalObject = extern struct {
         return err;
     }
 
+    pub fn createError(
+        this: *JSGlobalObject,
+        comptime code: JSC.Node.ErrorCode,
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.zig]: only `message` and `args` needs to be comptime

File: src/bun.js/webcore.zig [hunk 2]
```diff
@@ -537,19 +477,59 @@ pub const Crypto = struct {
...
+                maxmem,
+                null,
+                0,
+            ) != 1)) {
+                return throwInvalidParams(globalThis, "RangeError", "Invalid scrypt params\n", .{});
+            }
+
+            var slice: []u8 = undefined;
+            slice.len = 0;
+            return JSC.ArrayBuffer.create(globalThis, slice, .ArrayBuffer);
```
Comment (Jarred-Sumner) [inline src/bun.js/webcore.zig]: ?

Comment (Jarred-Sumner) [reply src/bun.js/webcore.zig]: oh

Can you create a special function for returning an empty ArrayBuffer? this looks really sketch / a mistake as-is

Comment (dylan-conway) [reply src/bun.js/webcore.zig]: 👍

File: src/bun.js/bindings/bindings.zig (hunk 1 above)
Comment (dylan-conway) [reply src/bun.js/bindings/bindings.zig]: 👍

File: src/bun.js/bindings/bindings.zig [hunk 3]
```diff
@@ -2414,6 +2414,19 @@ pub const JSGlobalObject = extern struct {
         return err;
     }
 
+    pub fn createError(
+        this: *JSGlobalObject,
+        code: JSC.Node.ErrorCode,
+        comptime error_name: string,
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.zig]: error_name doesnt need to be comptime either

Comment (dylan-conway) [reply src/bun.js/bindings/bindings.zig]: oops

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**